api/src/main/java/com/epam/pipeline/manager/cluster/autoscale/ReassignHandler.java

```diff
@@ -228,4 +235,12 @@ private boolean isValueTrue(final String value) {
         }
         return BooleanUtils.toBoolean(value);
     }
+
+    private Map<String, String> findCustomInstanceTags(final Long runId, final Optional<PipelineRun> optionalRun) {
+        final Optional<PipelineRun> run = optionalRun.isPresent() ? optionalRun : pipelineRunManager.findRun(runId);
+        if (!run.isPresent()) {
+            return Collections.emptyMap();
+        }
+        return metadataManager.prepareCustomInstanceTags(run.get());
```

Comment:
We can use more optional-friendly implementation here.

```suggestion
        return optionalRun.map(Optional::of)
                .orElseGet(() -> pipelineRunManager.findRun(runId))
                .map(metadataManager::prepareCustomInstanceTags)
                .orElseGet(Collections::emptyMap);
```